cpp/src/glm/ridge.cuh

```diff
@@ -187,7 +188,19 @@ void ridgeFit(const raft::handle_t& handle,
                    norm2_input.data(),
                    fit_intercept,
                    normalize,
-                   stream);
+                   sample_weight);
+  }
+  if (sample_weight != nullptr) {
+    raft::linalg::sqrt(sample_weight, sample_weight, n_rows, stream);
```

Comment:
We should add a note in the docs that `sample_weight` is going to be modified and it's possible that it might not be exactly the same after running the algorithm. Ideally it could be const and copied but I do see the benefit in trying to use it directly.